configList.go

```diff
@@ -0,0 +1,60 @@
+package notmuch
+
+// Copyright © 2015 The go.notmuch Authors. Authors can be found in the AUTHORS file.
+// Licensed under the GPLv3 or later.
+// See COPYING at the root of the repository for details.
+
+// #cgo LDFLAGS: -lnotmuch
+// #include <stdlib.h>
+// #include <notmuch.h>
+import "C"
+
+// Messages represents a notmuch config list
+type ConfigList cStruct
+
+//ConfigOption is used as a container for key / value pairs of the ConfigList
+type ConfigOption struct {
+	Key, Value string
+}
```

Comment:
Rather than declaring an extra struct to carry these, let's just make the `key` and `value` methods public, and get rid of the argument to next.

Reply:
I did this on purpose, to keep the api the same between Threads, messages and the config.
My first approach was actually just on the configlist directly.

I'll change it back if you prefer the Key/Val methods directly on the list.
How do you want to handle errors? I can either return an error or simply return the empty string

Reply:
Let's just return the empty string.

Reply:
Ack, just realized a problem with my suggestion: If you just do Key/Value inside the loop you skip the first element.

Let's go back to the old way, but instead of declaring a one-off struct, just give next two parameters:

```go
func Next(key, value *string) bool
```

Sorry I'm so disorganized.

(I saw your email. It's not obvious to me at a glance why it's segfaulting, but making this change will hopefully dodge the issue).